tests/unit/ng/connector/base.py

```diff
@@ -890,6 +890,46 @@ def test_pipeline_preprocessing_does_not_add_timestamp_delta_if_configured_but_l
 
             connector.shut_down()
 
+    @pytest.mark.parametrize(
+        ["timestamp", "expected_error_message"],
+        [
+            ("0000-00-00 00:00:00", "year 0 is out of range"),
+            ("invalid", "Invalid isoformat string: 'invalid'"),
+        ],
+    )
+    def test_pipeline_preprocessing_raises_criticalinputerror_on_timeparser_error(
```

Comment:
the test suggests that a criticalinputerror should be raised but I see no pytest.raises. consider to find a better name for this test.